core/src/main/java/com/segment/analytics/kotlin/core/platform/plugins/SegmentDestination.kt

```diff
@@ -95,8 +95,8 @@ class SegmentDestination : DestinationPlugin(), VersionedPlugin {
     override fun update(settings: Settings, type: Plugin.UpdateType) {
         super.update(settings, type)
         if (settings.hasIntegrationSettings(this)) {
-            settings.destinationSettings<SegmentSettings>(key)?.let {
-                pipeline.apiHost = it.apiHost
+            settings.destinationSettings<SegmentSettings>(key)?.apiHost?.let {
```

Comment:
OMG thats so subtle of a difference 😓 . Lets add a comment here indicating that we only want to populate the apiHost value if it exists